File: packages/agent-mux/watch-watchos-app/AgentMuxWatchApp/Transport/DirectGatewayClient.swift
import Foundation

final class DirectGatewayClient {
  private var task: URLSessionWebSocketTask?
  private let session = URLSession(configuration: .ephemeral)

  func connect(url: URL? = nil) {
    guard task == nil, let url else { return }
    task = session.webSocketTask(with: url)
    task?.resume()
  }

  func disconnect() {
    task?.cancel(with: .goingAway, reason: nil)
    task = nil
  }

  func send(_ text: String) {
    task?.send(.string(text)) { _ in }
  }
}

File: packages/agent-mux/watch-wearos-app/app/src/main/kotlin/ai/a5c/amux/wear/transport/DirectGatewayClient.kt
package ai.a5c.amux.wear.transport

class DirectGatewayClient {
  private var active = false

  fun connect() {
    active = true
  }

  fun disconnect() {
    active = false
  }

  fun send(payload: String) {
    if (!active) {
      connect()
    }
  }
}
